src/agent/src/cdh.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{anyhow, Result};
+use protocols::{
+    sealed_secret, sealed_secret_ttrpc_async, sealed_secret_ttrpc_async::SealedSecretServiceClient,
+};
+const CDH_ADDR: &str = "unix:///run/confidential-containers/cdh.sock";
+
+#[derive(Clone)]
+pub struct CDHClient {
+    sealed_secret_client: SealedSecretServiceClient,
+}
+
+impl CDHClient {
+    pub fn new() -> Result<Self> {
+        let c = ttrpc::asynchronous::Client::connect(CDH_ADDR)?;
+        let ssclient = sealed_secret_ttrpc_async::SealedSecretServiceClient::new(c);
+        Ok(CDHClient {
+            sealed_secret_client: ssclient,
+        })
+    }
+
+    pub async fn unseal_secret_async(
+        &self,
+        sealed: &str,
+    ) -> Result<sealed_secret::UnsealSecretOutput> {
+        let secret = sealed
+            .strip_prefix("sealed.")
+            .ok_or(anyhow!("strip_prefix sealed. failed"))?;
```

Comment:
Please add quotes around "sealed.", otherwise a user would have trouble parsing this error message.

What about a message like `stripping "sealed." prefix failed` ?

Reply:
fixed, thanks.